ui/jobs/jobs.js

```diff
@@ -2108,6 +2169,132 @@ class Bars {
     this.gainEffectFuncMap[EffectId.CoeurlForm] = changeFormFunc;
   }
 
+  setupDrg() {
+    const highJumpBox = this.addProcBox({
+      id: 'drg-procs-highjump',
+      fgColor: 'drg-color-highjump',
+    });
+    [
+      kAbility.HighJump,
+      kAbility.Jump,
+    ].forEach((ability) => {
+      this.abilityFuncMap[ability] = () => {
+        highJumpBox.duration = 0;
+        highJumpBox.duration = 30;
+      };
+    });
+
+    const disembowelBox = this.addProcBox({
+      id: 'drg-procs-disembowel',
+      fgColor: 'drg-color-disembowel',
+    });
+    this.statChangeFuncMap['DRG'] = () => {
+      disembowelBox.valuescale = this.gcdSkill();
+      disembowelBox.threshold = this.gcdSkill() * 5;
+      highJumpBox.valuescale = this.gcdSkill();
+      highJumpBox.threshold = this.gcdSkill() + 1;
+    };
+    // ComboTimer Bar
+    // DRG's skill effect buff appear on self at once,
+    // but start counting down when damage is deal.
+    // This cause some timer not accurate.
+    const comboTimer = this.addTimerBar({
+      id: 'drg-timers-combo',
+      fgColor: 'drg-color-combo',
+    });
+    let comboType = '';
+    const buffComboFunc = () => {
+      comboTimer.duration = 0;
+      comboTimer.duration = 10;
+      comboType = 'buff';
+    };
+    const comboLostFunc = () => {
+      comboTimer.duration = 0;
+    };
+
+    this.comboFuncs.push((skill) => {
+      // if this.combo.considerNext is equal as this.combo.startList, the combo is broken.
+      if (this.combo.considerNext == this.combo.startList)
+        comboTimer.duration = 0;
```

Comment:
Can you expose some method in the combo tracker, e.g. `IsComboActive()` rather than comparing these two lists? These lists are implementation details too.

Reply:
I'm not sure you means add a method to combotracker to compare these two list instead of compare lists outside?